boa/src/builtins/string/mod.rs

```diff
@@ -722,7 +704,12 @@ impl String {
             None
```

Comment:
This part here is not convincing me. I know it's not part of this request, but I think we could refactor this to make it more clear. Something like:
```rust
let fill_string = args.get(1).map(|arg| ctx.to_string(arg)).transpose()?;
```

Reply:
The bottom comment will remove this also.